cloud/insight_object.go

```diff
@@ -0,0 +1,217 @@
+package cloud
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+
+	"github.com/andygrunwald/go-jira/v2/cloud/models/apps/insights"
+)
+
+// GetObject returns a single object by id
+// Reference: https://developer.atlassian.com/cloud/insight/rest/api-group-object/#api-object-id-get
+func (i *InsightService) GetObject(ctx context.Context, workspaceID, id string) (*insights.Object, error) {
+	apiEndPoint := fmt.Sprintf(`%s/jsm/insight/workspace/%s/v1/object/%s`, insightURL, workspaceID, id)
+
+	req, err := i.client.NewRequest(ctx, http.MethodGet, apiEndPoint, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	req.Header.Set("Accept", "application/json")
+
+	res, err := i.client.client.Do(req)
+	if err != nil {
+		return nil, err
+	}
+
+	switch res.StatusCode {
+	case http.StatusUnauthorized:
+		return nil, fmt.Errorf("%s: %w", req.URL.String(), ErrUnauthorized)
+	case http.StatusNotFound:
+		return nil, fmt.Errorf("%s: %w", req.URL.String(), ErrNotFound)
+	case http.StatusInternalServerError:
+		return nil, fmt.Errorf("%s: %w", req.URL.String(), ErrUnknown)
+	}
+
+	object := new(insights.Object)
+	err = json.NewDecoder(res.Body).Decode(&object)
+
+	return object, err
+}
+
+// UpdateObject updates an object and returns it
+// Reference: https://developer.atlassian.com/cloud/insight/rest/api-group-object/#api-object-id-put
+func (i *InsightService) UpdateObject(ctx context.Context, workspaceID, id string, body insights.PutObject) (*insights.Object, error) {
+	apiEndPoint := fmt.Sprintf(`%s/jsm/insight/workspace/%s/v1/object/%s`, insightURL, workspaceID, id)
+
+	req, err := i.client.NewRequest(ctx, http.MethodGet, apiEndPoint, body)
+	if err != nil {
+		return nil, err
+	}
+
+	req.Header.Set("Accept", "application/json")
+
+	res, err := i.client.client.Do(req)
+	if err != nil {
+		return nil, err
+	}
+
+	switch res.StatusCode {
```

Comment:
The insight api uses the same response handling as jira api, so the idea was to make it more lightweight and easier to handle